--- a/mods/photonicat-pm.c
+++ b/mods/photonicat-pm.c
@@ -800,8 +800,8 @@ static int pcat_pm_uart_serdev_open(struct pcat_pm_data *pm_data)
 	sched_set_fifo(pm_data->kworker->task);
 		
 	kthread_init_work(&pm_data->check_work, pcat_pm_check_work);
-	hrtimer_init(&pm_data->check_timer, CLOCK_MONOTONIC, HRTIMER_MODE_REL_HARD);
-	pm_data->check_timer.function = pcat_pm_check_timer_expired;
+	hrtimer_setup(&pm_data->check_timer, pcat_pm_check_timer_expired,
+		      CLOCK_MONOTONIC, HRTIMER_MODE_REL_HARD);
 	
 	hrtimer_start(&pm_data->check_timer, ms_to_ktime(1000), HRTIMER_MODE_REL_HARD);
 	
@@ -837,7 +837,7 @@ static int pcat_pm_charger_probe(struct pcat_pm_data *pm_data)
 	}
 	
 	pscfg.drv_data = pm_data;
-	pscfg.of_node = charger_node;
+	pscfg.fwnode = of_fwnode_handle(charger_node);
 
 	pm_data->battery_psy = power_supply_register(dev, battery_psy_desc, &pscfg);
 	if (IS_ERR(pm_data->battery_psy)) {
